### PR TITLE
[PH] Provide *PluginArgs dataclasses to capture and validate nodeos config options

### PR DESCRIPTION
**Pull request by oschwaldp-oci**
Provide `*PluginArgs` dataclasses to capture and validate `nodeos` config options.

Each `*PluginArgs` dataclass captures the command line argument and default for each option as specified in `nodeos`.

New test `validate_nodeos_plugin_args.py` uses `nodeos`'s `--help` command to parse out each plugin's available options and defaults to validate current status vs. the `*PluginArgs` classes.  Any addition, removal, update will cause a test failure with note of what has changed.  This will allow easy maintenance going forward.

Consequently, quietly defaulted `nodeos` arguments will now have representation in the performance harness test reports.

### CODE REVIEW

File: tests/performance_tests/ChainPluginArgs.py [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+    _terminateAtBlockNodeosDefault: int=0
+    _terminateAtBlockNodeosArg: str="--terminate-at-block"
+    snapshot: str=None
+    _snapshotNodeosDefault: str=None
+    _snapshotNodeosArg: str="--snapshot"
+
+    def threads(self, threads: int):
+        self.chainThreads=threads
+
+    def supportedNodeosArgs(self) -> list:
```
Comment (jgiszczak) [inline tests/performance_tests/ChainPluginArgs.py:178]: This and `__str__()` are wonderfully generic.  I think it's time to create a parent class for all the PluginArgs family, so there's just one copy of these two methods.

File: tests/performance_tests/validate_nodeos_plugin_args.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    result = subprocess.run(["programs/nodeos/nodeos", "--help"], capture_output=True, text=True)
+
+    myStr = result.stdout
+    myStr = myStr.rstrip("\n")
+    myStr = re.sub(":\n\s+-",':@@@\n  -', string=myStr)
+    myStr = re.sub("\n\n",'\n@@@', string=myStr)
+    myStr = re.sub("Application Options:\n",'', string=myStr)
+    pluginSections = re.split("(@@@.*?@@@\n)", string=myStr)
+
+    sec=0
```
Comment (jgiszczak) [inline tests/performance_tests/validate_nodeos_plugin_args.py:28]: Surely this is just `len(pluginSections)`...

**Review by jgiszczak — CHANGES_REQUESTED** [review]

File: tests/performance_tests/ChainPluginArgs.py (hunk 1 above)
Comment (oschwaldp-oci) [reply tests/performance_tests/ChainPluginArgs.py]: Addressed: https://github.com/AntelopeIO/leap/pull/546/commits/0cd997ff6e547526afdf7379c2b911cdfd7d0113

File: tests/performance_tests/NodeosPluginArgs/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,11 @@
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/__init__.py ${CMAKE_CURRENT_BINARY_DIR}/__init__.py COPYONLY)
```
Comment (jgiszczak) [inline tests/performance_tests/NodeosPluginArgs/CMakeLists.txt:1]: Default behavior for `configure_file()`  is current source to current binary directory, so for new CMake files, we're omitting the redundant paths.  See `TestHarness` CMakeLists.txt for example.

File: tests/performance_tests/NodeosPluginArgs/generate_nodeos_plugin_args_class_files.py [hunk 4]
```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python3
```
Comment (jgiszczak) [inline tests/performance_tests/NodeosPluginArgs/generate_nodeos_plugin_args_class_files.py:1]: A file-level triple-quoted comment block describing the purpose and limitations of this script would be helpful.  It's easy enough to guess what it's for from the name I suppose, but a brief description of the output is probably beneficial.

**Review by jgiszczak — CHANGES_REQUESTED** [review]

File: tests/performance_tests/NodeosPluginArgs/CMakeLists.txt (hunk 3 above)
Comment (oschwaldp-oci) [reply tests/performance_tests/NodeosPluginArgs/CMakeLists.txt]: Addressed here: https://github.com/AntelopeIO/leap/pull/546/commits/52b7903d92b667fd836e97aa649c3021ba4c3f72

File: tests/performance_tests/NodeosPluginArgs/generate_nodeos_plugin_args_class_files.py (hunk 4 above)
Comment (oschwaldp-oci) [reply tests/performance_tests/NodeosPluginArgs/generate_nodeos_plugin_args_class_files.py:1]: Addressed here: [792a0f8](https://github.com/AntelopeIO/leap/pull/546/commits/792a0f8559656403d63b8c87ee952ff6b4cee140)

Also added a suggestion around using the script to regenerate when validation test fails: [2e4495e](https://github.com/AntelopeIO/leap/pull/546/commits/2e4495e651a9df196a0dd5710c854b64aa585e68)

**Review by jgiszczak — APPROVED** [review]

**Pull request merged**